src/languages/cpp.js

```diff
@@ -425,11 +425,7 @@ export default function(hljs) {
       _hint: FUNCTION_HINTS },
     begin: regex.concat(
       /\b/,
-      /(?!decltype)/,
-      /(?!if)/,
-      /(?!for)/,
-      /(?!switch)/,
-      /(?!while)/,
+      RegExp(`(?!${RESERVED_KEYWORDS.join('|')})`),
```

Comment:
Pretty sure we don't need the RegExp since `concat` handles all that behind the scenes?  Otherwise looks good.